packages/core/src/main/Account.ts

```diff
@@ -119,11 +129,12 @@ const coreDefaultClient: ClientInfo = {
   model: '@wireapp/core',
 };
 
-export class Account extends EventEmitter {
+export class Account<T = unknown> extends EventEmitter {
   private readonly apiClient: APIClient;
   private readonly logger: logdown.Logger;
   private readonly createStore: CreateStoreFn;
   private storeEngine?: CRUDEngine;
+  private readonly secretsCrypto?: SecretCrypto<T>;
```

Comment:
i have a question about the usage of a generic here in general: 
from the code it seems like the type will either be of Uint8Array or EncryptedPayload (depending on the source). 

I am wondering if this will be extended (possibly) in the future or if there is someother reason to use a generic instead of a union?

Reply:
Since here we are from the point of view of a lib, we need to be as open as possible. 
One consumer could give a function that takes and returns a `string` or a `number` or any object that they want to use for encryption/decryption. We cannot really guess from the library perspective was will be used by the consumer.